src/components/coffeechat.ts

```diff
@@ -0,0 +1,398 @@
+import { Client } from 'discord.js-commando';
+import { Database } from 'sqlite';
+import { openDB } from './db';
+import _ from 'lodash';
+import { Person, stableMarriage } from 'stable-marriage';
+
+const COFFEE_ROLE_ID: string = process.env.COFFEE_ROLE_ID || '.';
+const TARGET_GUILD_ID: string = process.env.TARGET_GUILD_ID || '.';
+const RANDOM_ITERATIONS = 1000;
+
+interface historic_match {
+  first_user_id: string;
+  second_user_id: string;
+  match_date: string;
+}
+
+interface future_match {
+  first_user_id: string;
+  second_user_id: string;
+  week_id: number;
+}
+
+export const initCoffeeChatTables = async (db: Database): Promise<void> => {
+  //Database to store past matches, with TIMESTAMP being the time matches were written into DB
+  await db.run(
+    `
+        CREATE TABLE IF NOT EXISTS coffee_historic_matches (
+            first_user_id TEXT NOT NULL,
+            second_user_id TEXT NOT NULL,
+            match_date TIMESTAMP NOT NULL
+        )
+        `
+  );
+  //Database to store finished status of future weeks
+  await db.run(
+    `
+        CREATE TABLE IF NOT EXISTS coffee_week_status (
+            week_id INTEGER PRIMARY KEY,
+            finished BOOL DEFAULT FALSE
```

Comment:
I think it would still be possible to insert a null value despite the default being false. I think `finished BOOL NOT NULL DEFAULT FALSE` should work.